### PR TITLE
Improving api account creation functionality

### PR DESCRIPTION
**Pull request by bertrand-deriv**
### Improving api account creation functionality
In this PR, 
- I have done some improvements to the api account creation functions so that we can be able to re-run successfully the tests that use `c_createRealAccount` command in one cypress session. Before, it used to throw timeout error because no api calls were being made.
- I have refactored the function to use cypress command instead of playwright for **getting verification code** from the events url. If you check well, I'm using the new command `c_emailVerificationV2` but it's almost similar to `c_emailVerification` already existing.

### CODE REVIEW

File: cypress/e2e/api/createAccountAndLogin.cy.js [hunk 1]
```diff
@@ -1,13 +1,14 @@
...
     cy.log('<E2EOAuthUrl - beforeEach>' + Cypress.env('oAuthUrl'))
+    cy.c_visitResponsive('/')
     cy.c_createRealAccount()
     cy.c_login()
   })
 
-  it('should get the oauth url from API', () => {
-    cy.log('<E2EOAuthUrl - Test 1>' + Cypress.env('oAuthUrl'))
+  it('should set the oauth url after successful account creation and login', () => {
+    cy.log('<E2EOAuthUrl - Test >' + Cypress.env('oAuthUrl'))
```
Comment (markodonnell-deriv) [inline cypress/e2e/api/createAccountAndLogin.cy.js:12]: Might be a good idea to log some of the credentials data here.

File: cypress/support/commands/common.js [hunk 2]
```diff
@@ -392,16 +436,16 @@ Cypress.Commands.add('c_loadingCheck', () => {
...
-
-    // Updating Cypress environment variables with the new email
-    const currentCredentials = Cypress.env('credentials')
-    currentCredentials.test.masterUser.ID = email
-    Cypress.env('credentials', currentCredentials)
+  // Call Verify Email and then set the Verification code in env
+  cy.task('wsConnect')
+  cy.task('verifyEmailTask').then((accountEmail) => {
+    cy.c_emailVerificationV2('account_opening_new.html', accountEmail)
+    cy.task('createRealAccountTask').then(() => {
```
Comment (markodonnell-deriv) [inline cypress/support/commands/common.js:443]: Could we have a try catch finally here, just incase we get an exception as the WS needs to alway disconnect and should go in the finally..

**Review by markodonnell-deriv — CHANGES_REQUESTED** [review]
Just some minor comments. Good job.

File: cypress.config.js [hunk 3]
```diff
@@ -15,9 +22,39 @@ module.exports = defineConfig({
     chromeWebSecurity: false,
     setupNodeEvents(on, config) {
       on('task', {
+        wsConnect() {
+          // Check if there is an existing connection and close it if open
+          if (connection && connection.readyState === WebSocket.OPEN) {
```
Comment (abuzar-deriv) [inline cypress.config.js:27]: Can you please shorten it to following for brevity?
```suggestion
          if(connection?.readyState === WebSocket.OPEN) {
```

File: cypress.config.js [hunk 4]
```diff
@@ -15,9 +22,39 @@ module.exports = defineConfig({
...
+          );
+          connection.onopen = () => console.log('Connection opened successfully');
+          connection.onerror = error => console.error('Connection error:', error);
+
+          api = new DerivAPI({ connection });
+
+          return null;
+        },
+        wsDisconnect() {
+          if (connection && connection.readyState === WebSocket.OPEN) {
```
Comment (abuzar-deriv) [inline cypress.config.js:43]: Same here:
```suggestion
          if (connection?.readyState === WebSocket.OPEN) {
```

**Review by abuzar-deriv — COMMENTED** [review]
Some minor comments. Looks great overall 💯

File: cypress/e2e/api/createAccountAndLogin.cy.js (hunk 1 above)
Comment (bertrand-deriv) [reply cypress/e2e/api/createAccountAndLogin.cy.js:12]: Which ones would you recommend to log ? like an email ??

File: cypress/support/commands/common.js (hunk 2 above)
Comment (bertrand-deriv) [reply cypress/support/commands/common.js:443]: yeah
That's right. Let me do it

**Review by markodonnell-deriv — APPROVED** [review]
Looks good

**Review by abuzar-deriv — APPROVED** [review]

**Pull request merged**